modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/GenomeInfo.pm

```diff
@@ -0,0 +1,76 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2020] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::Production::Pipeline::Webdatafile::GenomeInfo;
+
+use strict;
+use warnings;
+
+use base qw/Bio::EnsEMBL::Production::Pipeline::Common::Base/;
+use Bio::EnsEMBL::Registry;
+use Bio::EnsEMBL::Production::Search::GenomeFetcher;
+
+
+sub run {
+	my ( $self) = @_;
+        my $species = $self->param('species');
+	if ( $species !~ /Ancestral sequences/ ) {
+		my $compara = $self->param('compara');
+		if ( !defined $compara ) {
+			$compara = $self->division();
+			if ( !defined $compara || $compara eq '' ) {
+				$compara = 'multi';
+			}
+		}
+		my %genome = %{$self->genome_info( $species, $compara )};
+		# figure out output
+		$self->dataflow_output_id( {  
+                        species    => $species,
+			group      => 'core',
+                        genome_id  => $species . '_' . $genome{'assembly'}{'accession'},
+                        gca        => $genome{'assembly'}{'accession'},
+                        level      => $genome{'assembly'}{'level'},
+                        assembly_default => $genome{'assembly'}{'default'},       
+                        assembly_ucsc => $genome{'assembly'}{'ucsc'},       
+                        dbname     => $genome{'dbname'},
+                        version    => $genome{'assembly'}{'name'},  
+                        division   => $genome{'division'},
+                       species_id  => $genome{'species_id'},         
+                       root_path   => $self->param('output_path'),   
+                }, 1 );
+
+	} 
+} ## end sub dump
+
+sub genome_info {
+	my ( $self, $species, $compara ) = @_;
+	my $genome;
+	if ( $compara && $compara ne 'multi' ) {
```

Comment:
There's a fair amount of code in this module to do with the `$compara` variable, but it only seems to be used to determine whether ENS_VERSION and EG_VERSION should be passed to the `new` method (which is done when `ne` 'multi', which is equivalent to the division not being 'vertebrates' - that would seem a simpler test?) And why not pass the *_VERSION variables for vertbrates anyway?

Reply:
in GenomeFetcher module : 
i am setting 
$self->{info_adaptor}->set_ensembl_genomes_release($eg_version);
and 
$self->{info_adaptor}->set_ensembl_release($ens_version);
 
to get desired genome info for given release

Reply:
Yes; but my point is that you have a conditional structure here to decide whether to pass the eg_version and ens_version parameters to the `new` method of GenomeFetcher. That conditional is based on whether the compara parameter is "multi". This is just a proxy for the division being "vertebrates", so a) why not simply test the division in the conditional statement? and b) why only pass the version parameters for non-vertebrates?

Reply:
i followed conditional statement similar to GenomeFetcher.
a)  yes your right we can make it simple conditional statement by checking ENS or EG version but currently we run the pipeline both vert and nonvert species together so i followed same condition as GenomeFetcher 
b) mostly i did testing with vert species so i carried away. I will change the code accordingly